internal/nix/input.go

```diff
@@ -73,7 +73,15 @@ func PackageFromString(raw string, locker lock.Locker) *Package {
 		}
 		pkgURL, _ = url.Parse(normalizedURL)
 	}
-	return &Package{*pkgURL, locker, raw, ""}
+
+	pkg := &Package{
+		URL:                                 *pkgURL,
+		lockfile:                            locker,
+		Raw:                                 raw,
+		normalizedPackageAttributePathCache: "",
+	}
+
+	return pkg
```

Comment:
nit, you can revert this if you want. You can also omit `normalizedPackageAttributePathCache` if you used name fields.